pkg/registry/apis/provisioning/repository/github.go

```diff
@@ -59,21 +59,24 @@ func NewGitHub(
 ) *githubRepository {
 	owner, repo, _ := parseOwnerRepo(config.Spec.GitHub.URL)
 	token := config.Spec.GitHub.Token
-	if token == "" {
+	if token == "" && secrets != nil {
 		decrypted, err := secrets.Decrypt(ctx, config.Spec.GitHub.EncryptedToken)
 		if err == nil {
 			token = string(decrypted)
 		}
 	}
-	return &githubRepository{
+	gh := &githubRepository{
 		config:     config,
-		gh:         factory.New(ctx, token), // TODO, baseURL from config
 		secrets:    secrets,
 		webhookURL: webhookURL,
 		owner:      owner,
 		repo:       repo,
 		cloneFn:    cloneFn,
 	}
+	if factory != nil {
```

Comment:
what made you do these changes? we should always have a Github client for this Github repository.

Reply:
removed this -- now testing with a mock rather than direct implementation